WordPress/gutenberg
# Update layout / alignment metrics in the Site Editor's "Browse" mode.

## What?
Updates the general spacing of the UI in Browse mode.

## Why?
This is mostly an experiment for optimising the space for the content therein. Currently the sidebar occupies a large footprint with lofty spacing that may not scale well as more complex interfaces appear here. Indeed we're already running into minor issues with the readability of longer template descriptions. A before/after can be seen in the gif below:

<img src="https://user-images.githubusercontent.com/846565/224711083-b82c0918-6da0-4149-83fa-949471c3851b.gif" width="400">

Most significant changes:

* Site Icon and Save button now align with the frame †
* Equal left/right padding so that panel contents are centrally aligned on the Y axis
* Buttons are now the same size in all panels
* Reduced the size of `subheadline` in the experimental `Text` component
* Reduce visual prominence of the `<` button

† In order for this to work I had to increase the W button dimensions to 64px x 64px. The result is a small animation between browse and edit modes, since the W button needs to be 60x60 in edit mode to fit nicely in the top bar. I think we can discuss this, and personally I'd lean towards increasing the entire top bar to 64px in edit mode. That would be something to tackle separately, and the small animation might serve as a temporary solution.

Here's a gif demonstrating the layout formulation:

<img src="https://user-images.githubusercontent.com/846565/224712388-ff2c78e8-3862-43e7-b88d-4e7c22ed8a5d.gif" width="400">

Clearly there are a lot of changes here. In an ideal world we might tackle each item piecemeal, but I think doing so would result in awkward inbetween states where things get worse before getting better. Since the changes here are 90% css, I think there's an argument to deploy them altogether.

One other thing we might consider here is the width of the sidebar itself. The tighter spacing in this PR creates scope for us to reduce that if desirable.

## How?
Mostly css changes.

## Testing Instructions
1. Open the Site Editor and navigate around

---

To reiterate: I see this as an i1 experiment, and certainly something to be refined.

cc @WordPress/gutenberg-design
